frontend/web/components/RolePermissionsList.tsx

```diff
@@ -1,33 +1,52 @@
-import React, { useState, forwardRef, useImperativeHandle } from 'react'
+import React, {
+  useState,
+  forwardRef,
+  useImperativeHandle,
+  Ref,
+  FC,
+} from 'react'
 import Icon from './Icon'
 import { EditPermissionsModal } from './EditPermissions'
 import {
   useGetRoleProjectPermissionsQuery,
   useGetRoleEnvironmentPermissionsQuery,
 } from 'common/services/useRolePermission'
 import Format from 'common/utils/format'
+import { PermissionLevel, Req } from 'common/types/requests'
+import { Role } from 'common/types/responses'
 
-type MainItem = {
+type NameAndId = {
   name: string
-  id: string
+  id: number | string
+  [key: string]: any
 }
 
-type CollapsibleNestedRolePermissionsListProps = {
-  mainItems: MainItem[]
+type RolePermissionsListProps = {
+  mainItems: NameAndId[]
   role: Role
+  ref?: Ref<any>
   level: string
   filter: string
 }
 
-const PermissionsSummary = ({ level, levelId, role }) => {
+export type PermissionsSummaryType = {
+  level: PermissionLevel
+  levelId: number
+  role: Role
+}
```

Comment:
Why do we need to interleave camelCase when all the other types were snake_case?

Reply:
These don't get sent as request body and JavaScript prefers camel case